File: py/Pascal's Triangle.py
class Solution:
    # @return a list of lists of integers
    def generate(self, numRows):
        if numRows == 0:
            return [];
        
        count = 1;
        lret = [];
        
        lret.append([1]);
        
        while count < numRows:
            l = [];
            c = 0;
            l.append(lret[count-1][c]);
            while c < count-1:
                l.append(lret[count-1][c]+lret[count-1][c+1]);
                c += 1;
            l.append(lret[count-1][c]);
            lret.append(l);
            count += 1;
        
        return lret;
            
            
        

File: cpp/digit_problem/Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int> > generate(int numRows) {
        vector<vector<int>> vvret;
        vector<int> one(1,1);
        
        if(numRows == 0)
            return vvret;
            
        vvret.push_back(one);
        
        for(int i = 1;i < numRows;i++)
        {
            vector<int> cur;
            
            for(int j = 0;j < i+1;j++)
            {
                if(j == 0)
                    cur.push_back(1);
                else if(j == i)
                    cur.push_back(1);
                else
                {
                    cur.push_back(vvret.back()[j] + vvret.back()[j-1]);
                }
            }
            
            vvret.push_back(cur);
        }
        
        return vvret;
    }
};